### ISSUE TITLE
Amass fails to build (graph.go - type *cayley.Handle has no field or method OptimizeIterator)

### ISSUE THREAD

**Issue by seanmarpo** [OWASP/Amass#205]
Within the last few days, Amass has stopped building with multiple failures in `graph.go`. This has broken the `Dockerfile` as well.
```
$ go get -u github.com/OWASP/Amass/...
# github.com/OWASP/Amass/amass/handlers
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:102:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:451:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:476:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:500:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:538:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:579:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:644:19: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:660:19: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:686:19: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:708:20: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
../../go/src/github.com/OWASP/Amass/amass/handlers/graph.go:708:20: too many errors
```
My golang version
```
$ go version
go version go1.12.6 darwin/amd64
```

**Comment by mikemadden42** [comment]
I'm seeing the same issue.

```
$ go get -u -ldflags '-s -w' github.com/OWASP/Amass/cmd/amass
# github.com/OWASP/Amass/amass/handlers
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:102:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:451:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:476:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:500:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:538:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:579:17: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:644:19: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:660:19: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:686:19: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:708:20: g.store.OptimizeIterator undefined (type *cayley.Handle has no field or method OptimizeIterator)
go/src/github.com/OWASP/Amass/amass/handlers/graph.go:708:20: too many errors
```

``` 
$ go version
go version go1.12.6 darwin/amd64
```

**Comment by caffix** [comment]
If you build with support for modules turned on, then it should work.

export GO111MODULE=on

I’ll have to look into the Dockerfile, and of course we will fix the problem with the dependency in the next release.

**Comment by c0rvax** [comment]
@caffix 

I have just tested and `docker build -t amass https://github.com/OWASP/Amass.git` fails with the same error. Although, Dockerfile has this variable defined https://github.com/OWASP/Amass/blob/master/Dockerfile#L5

So probably the problem is somewhere else

**Comment by caffix** [comment]
I think the problem with the Dockerfile could just be that the environment variable needs to be defined earlier in the process. If the build was using the correct version of the dependency package, then there would not be a problem. Go Modules should ensure that the correct version is used

**Comment by c0rvax** [comment]
Okay, thanks. Feel free to close the pull request if you believe it is not necessary at the moment. Thanks for the quick updates!

**Comment by caffix** [comment]
The problem should be resolved in Amass version 3.0.17
